app/models/mixins/assignment_mixin.rb

```diff
@@ -128,44 +127,20 @@ def get_assigned_for_target(target, options = {})
       tlist =  parents.collect { |p| "#{p.class.base_model.name.underscore}/id/#{p.id}" } # Assigned directly to parents
       tlist += options[:tag_list] if options[:tag_list]                        # Assigned to target (passed in)
 
-      if options[:associations_preloaded]
-        # Collect tags directly from association from parent objects if they were already preloaded by the caller
-        tags = parents.collect { |p| p.tags.select { |t| t.name.starts_with?("/managed/") } }.flatten.uniq
-      else
-        # Collect tags from all parent objects in a single query if they were NOT already preloaded by the caller
-        tcond = []; targs = []
-        parents.each do |p|
-          tcond << "(taggings.taggable_type=? AND taggings.taggable_id=?)"
-          # TODO: we may need to change taggings-related code to use base_model too
-          targs << p.class.base_class.name << p.id
-        end
-        cond = ["(#{tcond.join(" OR ")}) AND (name like '/managed/%')", *targs]
-        tags = Tag.where(cond).joins(:taggings)
-      end
-      # Assigned to parent tags
-      # TODO: we may need to change taggings-related code to use base_model too
-      parent_ids_by_type = parents.inject({}) { |h, p|  h[p.class.base_class.name] ||= []; h[p.class.base_class.name] << p.id; h }
-      tlist += tags.inject([]) do |arr, tag|
-        tag.taggings.each do |t|
-          # Only collect taggings for parent objects
-          klass = t.taggable_type
-          if parent_ids_by_type[klass] && parent_ids_by_type[klass].include?(t.taggable_id)
-            if klass == "VmOrTemplate"       # right now NO support for tagged templates
-              arr << "vm/tag#{tag.name}"
-            else
-              arr << "#{klass.underscore}/tag#{tag.name}"
-            end
-          end
-        end
-        arr
-      end
+      individually_assigned_alerts = alist.select { |a| tlist.include?(a[:assigned_to]) }.map { |a| a[:assigned] }
 
-      result = alist.inject([]) do |arr, a|
-        arr << a[:assigned] if tlist.include?(a[:assigned_to])
-        arr
+      # look for alert_set running off of tags (not individual tags)
+      # TODO: we may need to change taggings-related code to use base_model too
+      tlist = Tagging.where("tags.name like '/managed/%'")
```

Comment:
ok, 

1. Making `assigned` into a hash makes the `to_set` unnecessary, since the array would drive the loop rather than `assigned` (aka `alist`). the code would just loop over the array. I'll put in a followup pr for this. The cache is used by one other part of the code.
2. Phew. Looks like putting `options[:tag_list]` into the first `tlist` or the second makes no difference. The code looks those values up in the `assigned` (aka `alist`) either way.